src/util/fork_utils.js

```diff
@@ -101,6 +109,25 @@ function nsfs_io_stats_handler(msg) {
     }
 }
 
+/**
+ * Sends a message to the worker to start the fork server with the giver port
+ * NOTE - currently runs only on non containerized enviorment
+ * @param {*} worker
+ * @param {*} offset
+ * @param {*} nsfs_config_root
+ */
+async function _send_fork_server_message(worker, offset, nsfs_config_root) {
+    const is_nc = process.env.NC_NSFS_NO_DB_ENV && process.env.NC_NSFS_NO_DB_ENV === 'true';
+    if (is_nc && offset >= 0) {
+        //wait for the worker to be ready to receive messages
+        await once(worker, 'message');
```

Comment:
add timeout so we won't wait forever?

Reply:
added 5 minute timeout. its worth mentioning that if the fork is killed the listener is canceled and the code will not run entirely